stats/stats.go

```diff
@@ -0,0 +1,141 @@
+package stats
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/elliotchance/orderedmap"
+	"github.com/hashicorp/go-hclog"
+	"github.com/segmentio/stats/v4"
+)
+
+type stat struct {
+	Start    time.Time
+	Duration time.Duration
+}
+
+type logHandler struct {
+	logger            hclog.Logger
+	trackedOperations *orderedmap.OrderedMap
+	mu                sync.Mutex
+}
+
+type Options struct {
+	tick    time.Duration
+	handler stats.Handler
+}
+
+func NewClockWithObserve(name string, tags ...stats.Tag) *stats.Clock {
+	// The default clock doesn't send a measurement on start (only on stop)
+	// We want both on start AND stop, so we wrap the ClockAt method
+	cl := stats.DefaultEngine.ClockAt(name, time.Now(), tags...)
+	stats.DefaultEngine.Observe(name, time.Duration(0), tags...)
+	return cl
+}
+
+func meta(name string, tags []stats.Tag) (string, bool) {
```

Comment:
please sort the private and public methods in the file.

Reply:
Done in https://github.com/cloudquery/cq-provider-sdk/pull/237/commits/fc1fb250a852284e19df964d65da57165d5de25a